pkg/splunk/controller/statefulset.go

```diff
@@ -112,7 +221,23 @@ func UpdateStatefulSetPods(c splcommon.ControllerClient, statefulSet *appsv1.Sta
 		// scale up StatefulSet to match desiredReplicas
 		scopedLog.Info("Scaling replicas up", "replicas", desiredReplicas)
 		*statefulSet.Spec.Replicas = desiredReplicas
-		return splcommon.PhaseScalingUp, splutil.UpdateResource(c, statefulSet)
+		err := splutil.UpdateResource(c, statefulSet)
+		if err != nil {
+			scopedLog.Error(err, "Unable to update statefulset")
+			return splcommon.PhaseError, err
+		}
+
+		// Check if the revision was updated successfully for Statefulset.
+		// It so can happen that it may take few seconds for the update to be
+		// reflected in the resource. In that case, just return from here and
+		// check the status back in the next reconcile loop.
```

Comment:
nit: "It can happen"
If this check can fail and its not big deal, we'll just check it at the next reconcile, do we need it?

Reply:
During the next reconcile we'll landup either in line 196 or 208, since we have already updated `replicas` here in this reconcile.